lib/kvdb/kvdb_meta.c

```diff
@@ -47,11 +46,12 @@ kvdb_meta_open(const char *const kvdb_home, FILE **meta_file)
     if (n < 0)
         return merr(EBADMSG);
 
-    meta_fd = open(buf, O_RDWR | O_SYNC, KVDB_META_PERMS);
+    flags = rdonly ? O_RDONLY : O_RDWR | O_SYNC;
+    meta_fd = open(buf, flags, KVDB_META_PERMS);
     if (meta_fd == -1)
         return merr(errno);
 
-    *meta_file = fdopen(meta_fd, "r+");
+    *meta_file = fdopen(meta_fd, rdonly ? "r" : "r+");
```

Comment:
Thoughts on how to automate tests for running on read-only storage devices?

Reply:
Nabeel indicated the smoke test verifies functionality on read files, but not read only filesystems.